core/trino-spi/src/main/java/io/trino/spi/connector/MergeDetails.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class MergeDetails
+{
+    public static final int DEFAULT_CASE_OPERATION_NUMBER = -1;
```

Comment:
Should this be filtered out in the engine so that connectors never see it?

Reply:
I'm not sure, but I'll think about it.

Reply:
I've removed DEFAULT_CASE_OPERATION_NUMBER from the SPI, and changed the merge processors to filter out the rows that have operation = DEFAULT_CASE_OPERATION_NUMBER.